include/nlohmann/json.hpp

```diff
@@ -3727,20 +3796,72 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
     with range checking
     @sa see @ref value() for access by value with a default value
 
-    @since version 1.0.0
+    @since version 1.0.0; template added in version 3.10.0
     */
-    const_reference operator[](const typename object_t::key_type& key) const
+    template < class KeyT, typename std::enable_if <
+                   !std::is_same<typename std::decay<KeyT>::type, json_pointer>::value&&
+#if defined(JSON_HAS_CPP_17)
+                   std::is_same<typename std::decay<KeyT>::type, std::string_view>::value&&
+#endif
+                   std::is_convertible<KeyT, typename object_t::key_type>::value, int >::type    = 0 >
+    const_reference operator[](KeyT && key) const
     {
         // const operator[] only works for objects
         if (JSON_HEDLEY_LIKELY(is_object()))
         {
-            JSON_ASSERT(m_value.object->find(key) != m_value.object->end());
-            return m_value.object->find(key)->second;
+            auto it = m_value.object->find(std::forward<KeyT>(key));
+            JSON_ASSERT(it != m_value.object->end());
```

Comment:
is an assert the right operation? Seems like an exception would be closer to the behavior one would expect from std::map ?

Reply:
No, it's a precondition that `key` exists. In release mode, there should be no runtime check for this.